crates/wasi-common/src/sys/windows/hostcalls_impl/fs.rs

```diff
@@ -197,11 +232,11 @@ fn file_access_mode_from_fdflags(
     let mut access_mode = AccessMode::READ_CONTROL;
 
     if read {
-        access_mode.insert(AccessMode::GENERIC_READ);
+        access_mode.insert(AccessMode::FILE_GENERIC_READ);
```

Comment:
What is the difference between those two?

Reply:
> What is the difference between those two?

`GENERIC_READ` is a special bit that can be set in the [access mask](https://docs.microsoft.com/en-us/windows/win32/secauthz/access-mask-format) and functionally is equivalent to setting _all_ bits for [`STANDARD_READ_RIGHTS` which `FILE_GENERIC_READ` is part of](https://docs.microsoft.com/en-us/windows-hardware/drivers/kernel/access-mask). So, as such, `FILE_GENERIC_READ` is more fine-grained control, and I guess it can avoid silly errors when we set the special bit just to find out that in a different syscall we're actually checking for the finer-grained `FILE_GENERIC_READ` which will not necessarily be set while we still have granted all the required rights with the special bit. Weirdness...

Reply:
That's correct.  We originally had `FILE_GENERIC_READ` and `FILE_GENERIC_WRITE`, which I changed to `GENERIC_READ` and `GENERIC_WRITE`, respectively, because `GENERIC_WRITE` is required to  open a file for truncation.

However, that broke append-only file mode as the `GENERIC_WRITE` access will override turning off the `FILE_WRITE_DATA` access that is required to treat a file as append-only.

Thus, this switches the implementation back to the less generic access modes so we can explicitly set `GENERIC_WRITE` only for truncation.

Reply:
Sounds fair enough! Please add a comment why we can't use `GENERIC_READ` and `GENERIC_WRITE`.

Reply:
Will do!